### PR TITLE
[k8s-configuration] Adding k8s-configuration extension, Deprecating k8sconfiguration

### PR DESCRIPTION
**Pull request by jonathan-innis**
This PR:
- Adds the deprecation message for the `k8sconfiguration` command group
- Adds the same configuration application logic over to `k8s-configuration` with some minor changes for GA release
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-configuration

**Comment by jonathan-innis** [comment]
@houk-ms @fengzhou-msft Please remove the `do-not-merge` label from this PR, it is now ready to merge

File: src/k8s-configuration/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[bdist_wheel]
+universal=1
```
Comment (fengzhou-msft) [inline src/k8s-configuration/setup.cfg:2]: Remove these 2 lines as we dropped support for Python 2.

**Review by houk-ms — APPROVED** [review]

File: src/k8s-configuration/setup.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'License :: OSI Approved :: MIT License',
+]
+
+# TODO: Add any additional SDK dependencies here
+DEPENDENCIES = ["pycryptodome~=3.9.8", "paramiko~=2.6.0", "cryptography>=2.3.1,<3.0.0"]
```
Comment (fengzhou-msft) [inline src/k8s-configuration/setup.py:36]: [paramiko](https://github.com/Azure/azure-cli/blob/1f875891e6cb858a1d02da8fede885ad7e348832/src/azure-cli-core/setup.py#L57) is a dependency of `azure-cli-core`, it should be removed here, then you don't need to pin `cryptography` which is a dependency of `paramiko`.

File: src/k8s-configuration/setup.cfg (hunk 1 above)
Comment (jonathan-innis) [reply src/k8s-configuration/setup.cfg]: Removed

File: src/k8s-configuration/setup.py (hunk 2 above)
Comment (jonathan-innis) [reply src/k8s-configuration/setup.py]: Removed dependencies

**Pull request merged**